src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs

```diff
@@ -176,7 +176,8 @@ public static class MicrosoftIdentityWebApiAuthenticationBuilderExtensions
                     // Process CIAM tenants
                     if (mergedOptions.Authority != null)
                     {
-                        mergedOptions.Authority = AuthorityHelpers.BuildCiamAuthorityIfNeeded(mergedOptions.Authority);
+                        mergedOptions.Authority = AuthorityHelpers.BuildCiamAuthorityIfNeeded(mergedOptions.Authority, out bool preserveAuthority);
+                        mergedOptions.PreserveAuthority = preserveAuthority;
```

Comment:
This code, the "orchestrator" knows about `preserveAuthority`

Reply:
It was related to renaming `BuildCiamAuthority" which was discussed. Closing this.